packages/react-popup/Spec.md

```diff
@@ -1,31 +1,385 @@
+- onPositioned
+- onScroll
+
+### Hidden mount
+
+v8 `Callout` provides two props which will allow mounting a hidden popup and disabling renders when the component is hidden. According to the PRs that introduced the features, it should be a performance optimization.
+
+- `hidden` [#4419](https://github.com/microsoft/fluentui/pull/4419)
+- `shouldUpdateWhenHidden` [#10465](https://github.com/microsoft/fluentui/pull/10465)
 
 ## Sample Code
 
-_Provide some representative example code that uses the proposed API for the component_
+```tsx
+<Popover>
+  <PopoverTrigger>
+    <button>Opens popup</button>
+  </PopoverTrigger>
+
+  <PopoverContent>
+    <h1>Popover</h2>
+    <div>Some section</div>
+    <div>Some section</div>
+  </PopoverContent>
+</Popover>
+```
 
 ## Variants
 
-_Describe visual or functional variants of this control, if applicable. For example, a slider could have a 2D variant._
+- Anchor content to a different target than the trigger
+- Inline popup rendered in DOM order
+- Open on:
+  - Click
+  - Hover
+  - Context menu (right click)
+  - Focus
 
 ## API
 
-_List the **Props** and **Slots** proposed for the component. Ideally this would just be a link to the component's `.types.ts` file_
+The `Popover` component will use React context to manage both a trigger and content component.
+
+### Popover
+
+Outer component that setups context and does not render DOM.
+
+> TODO Discuss: dismiss on scroll ? Seems to be a thing in v8 because callout doesn't support repositioning
+
+> TODO Discuss: v8 `hidden` `shouldUpdateHidden` prop [#4419](https://github.com/microsoft/fluentui/pull/4419) [#10465](https://github.com/microsoft/fluentui/pull/10465)
+
+> TODO Discuss: v8 `onPositioned`
+
+> TODO Discuss: v0/v8 `autoFocus/setInitialFocus` ? -> ***(not focus trap)** can be achieved pretty easily in userland with/without Tabster, in Teams this is extremely rare
+
+> TODO Discuss: A11y -> Should only one popup be open at a time or is `aria-hidden` enough ?
+
+> TODO Discuss: merge `position` and `align` props -> no real reason they were separated in v0 in the first place
+
+> TODO Discuss: start small with API base for positioning props ?
+
+The `@fluentui/react-positioning` library that exports the `usePopper` hook which will power the `Popover` contains more than the declared props here. These extra positioning props should be exposed as required.
+
+```typescript
+export interface PopoverProps {
+  /**
+   * Controls the popup open state
+   */
+  open?: boolean;
+
+  /**
+   * Call back when the component requests to change value
+   */
+  onOpenChange?: (e: OpenPopoverEvents, data: OpenEventData) => void;
+
+  /**
+   * Anchor the popup to an element other than the trigger
+   */
+  target?: HTMLElement;
+
+  /**
+   * Popover position relative to target
+   */
+  position?: string;
+
+  /**
+   * Popover alignment relative to target
+   */
+  align?: string;
```

Comment:
It might be good to constrain the type of `position` and `align` to a set of values and not only `string`. Also, should `align` be called `alignment`?

Here's a suggestion:

```suggestion
  /**
   * Popover render position relative to target
   */
  position?: 'after' | 'before' | 'bottom' | 'top';

  /**
   * Popover alignment to target based on its position
   */
  alignment?: 'start' | 'center' | 'end';
```

Reply:
v0 already uses `align` so it the familiar option, however we should discuss whether to merge the two tbh. Although I quite like that for RTL, only one needs to be flipped

Reply:
Added explicit values for the props